test/util/framework/per_test_framework.go

```diff
@@ -236,16 +263,67 @@ func (tc *perItOrDescribeTestContext) NewResourceGroup(ctx context.Context, reso
 // cleanupResourceGroup is the standard resourcegroup cleanup.  It attempts to
 // 1. delete all HCP clusters and wait for success
 // 2. delete the resource group and wait for success
-func cleanupResourceGroup(ctx context.Context, hcpClient *hcpsdk20240610preview.HcpOpenShiftClustersClient, resourceGroupsClient *armresources.ResourceGroupsClient, resourceGroupName string) error {
+func cleanupResourceGroup(ctx context.Context, hcpClient *hcpsdk20240610preview.HcpOpenShiftClustersClient, resourceGroupsClient *armresources.ResourceGroupsClient, resourceGroupName string, timeout time.Duration) error {
 	errs := []error{}
 
 	ginkgo.GinkgoLogr.Info("deleting all hcp clusters in resource group", "resourceGroup", resourceGroupName)
-	if err := DeleteAllHCPClusters(ctx, hcpClient, resourceGroupName, 60*time.Minute); err != nil {
+	if err := DeleteAllHCPClusters(ctx, hcpClient, resourceGroupName, timeout); err != nil {
 		return fmt.Errorf("failed to cleanup resource group: %w", err)
 	}
 
 	ginkgo.GinkgoLogr.Info("deleting resource group", "resourceGroup", resourceGroupName)
-	if err := DeleteResourceGroup(ctx, resourceGroupsClient, resourceGroupName, 60*time.Minute); err != nil {
+	if err := DeleteResourceGroup(ctx, resourceGroupsClient, resourceGroupName, false, timeout); err != nil {
+		return fmt.Errorf("failed to cleanup resource group: %w", err)
+	}
+
+	return errors.Join(errs...)
+}
+
+// cleanupResourceGroupNoRP performs cleanup when the resource provider is not available.
+// This is used to cleanup personal dev e2e test runs, where the infra is already gone so there's no
+// RP to call for HCP deletion.
+//  1. discovers any "managed" resource groups whose ManagedBy references a resource in the parent
+//     resource group and deletes them (using 'force' to speed up VM/VMSS deletion).
+//  2. deletes the parent resource group itself.
+func cleanupResourceGroupNoRP(ctx context.Context, resourceGroupsClient *armresources.ResourceGroupsClient, resourceGroupName string, timeout time.Duration) error {
+	errs := []error{}
+
+	// Discover managed resource groups by inspecting the ManagedBy field.
+	managedResourceGroups := []string{}
+	pager := resourceGroupsClient.NewListPager(nil)
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to list resource groups while discovering managed groups: %w", err)
+		}
+		for _, rg := range page.Value {
+			if rg.ManagedBy == nil {
+				continue
+			}
+
+			// Match managed resource groups whose owner is an HCP resource in the parent resource group.
+			if strings.Contains(
+				*rg.ManagedBy,
```

Comment:
I think you want `strings.ToLower()` on this - not guaranteed that an ARM resource ID is in any specific case